File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/MasterCrafter.java
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager;

/* JADX INFO: loaded from: classes3.dex */
public class MasterCrafter extends Quest {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void calculateTargetProgress(int i) {
        this.targetProgress = i - 1;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    protected void configure() {
        this.idName = R.string.quest_master_crafter_name;
        this.idDescription = R.string.quest_master_crafter_description;
        this.defaultRarity = 1;
        this.minimumDifficulty = 2;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void realignStaticReference() {
        QuestsManager.masterCrafter = this;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/MasterCrafter.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager

class MasterCrafter : Quest() {
    override fun calculateTargetProgress(difficulty: Int) {
        targetProgress = (difficulty - 1).toLong()
    }

    override fun configure() {
        idName = R.string.quest_master_crafter_name
        idDescription = R.string.quest_master_crafter_description
        defaultRarity = 1
        minimumDifficulty = 2
    }

    override fun realignStaticReference() {
        QuestsManager.masterCrafter = this
    }
}
